addons/test_mail/tests/test_mail_push.py

```diff
@@ -86,55 +62,75 @@ def _assert_notification_count_for_cron(self, number_of_notification):
         self.assertEqual(notification_count, number_of_notification)
 
     @patch.object(odoo.addons.mail.models.mail_thread, 'push_to_end_point')
-    def test_push_notifications(self, push_to_end_point):
-        # Test No Inbox Condition
-        self.record_simple.with_user(self.user_inbox).message_notify(
-            partner_ids=self.user_email.partner_id.ids,
-            body='Test',
-            subject='Test Activity',
-            record_name=self.record_simple._name,
-        )
-
-        self._assert_notification_count_for_cron(0)
-        push_to_end_point.assert_not_called()
-
-
-        self.record_simple.with_user(self.user_email).message_notify(
-            partner_ids=self.user_inbox.partner_id.ids,
-            body='Test message send via Web Push',
-            subject='Test Activity',
-            record_name=self.record_simple._name,
-        )
-
-        self._assert_notification_count_for_cron(0)
-        push_to_end_point.assert_called_once()
-        payload_value = json.loads(push_to_end_point.call_args.kwargs['payload'])
-        self.assertIn(self.record_simple._name, payload_value['title'])
-        self.assertIn(self.user_email.name, payload_value['title'])
-        self.assertEqual(payload_value['options']['body'], 'Test message send via Web Push')
-        self.assertEqual(payload_value['options']['data']['res_id'], self.record_simple.id)
-        self.assertEqual(payload_value['options']['data']['model'], self.record_simple._name)
-        self.assertIn('icon', payload_value['options'])
-        self.assertEqual(push_to_end_point.call_args.kwargs['device']['endpoint'], 'https://test.odoo.com/webpush/user2')
-        self.assertIn('vapid_private_key', push_to_end_point.call_args.kwargs)
-        self.assertIn('vapid_public_key', push_to_end_point.call_args.kwargs)
+    def test_notify_by_push(self, push_to_end_point):
+        """ In case of notification, only inbox users are notified """
+        for recipient, has_notification in [(self.user_email, False), (self.user_inbox, True)]:
+            with self.subTest(recipient=self.user_email):
```

Comment:
Ok but you force `recipient=self.user_email` instead of the one used in the current loop iteration though, which does not feel right, unless I misunderstood the concept of subTest.

Reply:
Ah yes.